src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/products.html.twig

```diff
@@ -74,60 +76,138 @@
-                 data-toggle="pstooltip"
-                 data-placement="top"
-                 data-original-title="{{ 'Edit'|trans({}, 'Admin.Actions') }}"
-              >
-                <i class="material-icons">edit</i>
-              </a>
+            {% if not orderForViewing.delivered %}
+              <td class="text-right">
+                <a href="#"
+                   class="tooltip-link"
+                   data-toggle="pstooltip"
+                   data-placement="top"
+                   data-original-title="{{ 'Edit'|trans({}, 'Admin.Actions') }}"
+                >
+                  <i class="material-icons js-order-product-update-btn"
+                     data-update-url="{{ path('admin_orders_update_product', {'orderId': orderForViewing.id, 'orderDetailId': product.orderDetailId}) }}"
+                     data-product-name="{{ product.name }}"
+                     data-product-quantity="{{ product.quantity }}"
+                     data-product-price-tax-incl="{{ product.unitPriceTaxInclRaw }}"
+                     data-product-price-tax-excl="{{ product.unitPriceTaxExclRaw }}"
+                     data-toggle="modal"
+                     data-target="#updateOrderProductModal"
+                  >edit</i>
+                </a>
 
-              <a href="#"
-                 class="ml-2 tooltip-link"
-                 data-toggle="pstooltip"
-                 data-placement="top"
-                 data-original-title="{{ 'Delete'|trans({}, 'Admin.Actions') }}"
-              >
-                <i class="material-icons">delete</i>
-              </a>
-            </td>
+                <a href="#"
+                   class="ml-2 tooltip-link"
+                   data-toggle="pstooltip"
+                   data-placement="top"
+                   data-original-title="{{ 'Delete'|trans({}, 'Admin.Actions') }}"
+                >
+                  <i class="material-icons">delete</i>
+                </a>
+              </td>
+            {% endif %}
           </tr>
         {% endfor %}
       </tbody>
     </table>
 
     <div class="row mb-3">
       <div class="col-md-12 text-right">
-        <button type="button" class="btn btn-outline-secondary">
-          <i class="material-icons">add_circle_outline</i>
-          {{ 'Add a product'|trans({}, 'Admin.Orderscustomers.Feature') }}
-        </button>
+        {% if not orderForViewing.delivered %}
+          <button type="button" class="btn btn-outline-secondary" data-toggle="modal" data-target="#addOrderProductModal">
+            <i class="material-icons">add_circle_outline</i>
+            {{ 'Add a product'|trans({}, 'Admin.Orderscustomers.Feature') }}
+          </button>
+        {% endif %}
         <button type="button" class="btn btn-outline-secondary" data-toggle="modal" data-target="#addOrderDiscountModal">
           <i class="material-icons">confirmation_number</i>
           {{ 'Add a new discount'|trans({}, 'Admin.Orderscustomers.Feature') }}
```

Comment:
Haven't we said ([here](https://github.com/PrestaShop/PrestaShop/pull/15388)) it should be _Add new cart rule_ (_Admin.Catalog.Feature_) instead?

Reply:
I can change, but "Discount" word is used a lot as well.

Reply:
Indeed but the change of wording has already been merged [here](https://github.com/PrestaShop/PrestaShop/pull/15761/files), I guess it means we cannot change it again.

Sorry @sarjon, I try to gather all feedback to avoid confusion, it's quite difficult to follow this page's migration, they are many pull requests to double-check.
😕

Reply:
We can change, it's not a problem and it wasn't released yet. :)